datasafe-types-api/src/test/java/de/adorsys/datasafe/types/api/shared/AwsClientRetry.java

```diff
@@ -0,0 +1,30 @@
+package de.adorsys.datasafe.types.api.shared;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AwsClientRetry {
+    @SneakyThrows
+    public static void createBucketWithRetry(AmazonS3 client, String bucket) {
+        try {
+            client.createBucket(bucket);
+        } catch (AmazonS3Exception ex) {
```

Comment:
Instead of parsing exception String, maybe we can simply use
```
  public static void createBucketWithRetry(AmazonS3 client, String bucket) {
        await().atMost(Duration.TEN_SECONDS).pollInterval(Duration.ONE_SECOND).untilAsserted(() -> client.createBucket(bucket));
}
```
That simply tries to create bucket each 1 second and fails if the bucket was not created within 10 seconds

Add awaitability library for that:
```
        <dependency>
            <groupId>org.awaitility</groupId>
            <artifactId>awaitility</artifactId>
        </dependency>
```

Reply:
This is a great tip. My retry block is not elegant but alog was visiable in the first successful build:
" we wait here for 3000 millis and retry due to eception :public class "
So I knew it is working and for that I added some logging to your approach.